### DISCUSSION TITLE
Restrict file fields to only take files from the library

### DISCUSSION THREAD

**Discussion by martijn-dev** [directus/directus#8999] [Feature Requests]
It would be nice to restrict a file field on a collection to only grap files from the file library.
So in edit mode I would only see "Choose file from Library" and not "Upload File from Device" and "Import File from URL".

This could be a setting on the file field, but depending on what might be more logic we could also forbid a specific role to upload files. Depending on the user role we can then hide out the options in the edit mode front-end.

Let me know what you guys think!

Cheers,
Martijn

**Comment by benhaynes** [comment]
Our other relational fields allow you to toggle add new and select existing, this seems to be the same thing, and I agree should be added. Seems like an interface option to me.

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
